src/cache_cf.cc

```diff
@@ -1853,17 +1838,8 @@ parse_http_header_replace(HeaderManglers **pm)
 static void
 dump_cachedir(StoreEntry * entry, const char *name, const Store::DiskConfig &swap)
 {
-    SwapDir *s;
-    int i;
-    assert (entry);
-
-    for (i = 0; i < swap.n_configured; ++i) {
-        s = dynamic_cast<SwapDir *>(swap.swapDirs[i].getRaw());
-        if (!s) continue;
-        storeAppendPrintf(entry, "%s %s %s", name, s->type(), s->path);
-        s->dump(*entry);
-        storeAppendPrintf(entry, "\n");
-    }
+    assert(entry);
+    Store::Disks::Dump(swap, *entry, name);
```

Comment:
The assert is unnecessary here.

Also since this is now just a one-liner calling a function from the legacy API dump*() name the whole static definition can be replaced by a #define dump_cachedir(X,Y,Z)  Store::Disks::Dump((Z), *(X), (Y)) in store/Disks.h alongside the free_cachedir() declaration.

Reply:
> the whole static definition can be replaced by a #define dump_cachedir(X,Y,Z) Store::Disks::Dump((Z), *(X), (Y)) in store/Disks.h alongside the free_cachedir() declaration.

A simple static definition inside a `.cc` file is much better than a `#define` in a public header. That free_cachedir() declaration was made public in c8f4eac for unit tests. If we are to clean this up further, we should model free_cachedir() after dump_cachedir() rather than half-mimic an already problematic free_cachedir() with a dump_cachedir() macro! However, I would not do that cleanup inside this PR.